src/qmapshack/gis/trk/filter/filter.cpp

```diff
@@ -655,6 +655,39 @@ void CGisItemTrk::filterLoopsCut(qreal minLoopLength)
     new CGisItemTrk(tr("%1 (Part %2)").arg(trk.name).arg(part), pts.first().idxTotal, pts.last().idxTotal, trk, project);
 }
 
+void CGisItemTrk::filterSplitTrack(qint8 nTracks)
+{
+    IGisProject* project = CGisWorkspace::self().selectProject(false);
+    if(nullptr == project)
+    {
+        return;
+    }
+    qint32 overlapNodes = 1;
+
+    qint32 segNodes = cntTotalPoints/nTracks+1;
+    qint32 segStartIdx = 0;
+    qint32 segEndIdx = segNodes-1;
+    qint32 remaining = cntTotalPoints;
+
+    qint8 part=0;
+    while(remaining>0)
+    {
+        new CGisItemTrk(tr("%1 (Part %2)").arg(trk.name).arg(part), segStartIdx, segEndIdx, trk, project);
+        // update remaining
+        segNodes = segEndIdx-segStartIdx+1;
```

Comment:
What is this good for? segNodes shouldn't change.  It's just the last segment that might differ in size depending on the remaining points.